src/shared/install.c

```diff
@@ -2527,11 +2523,17 @@ int unit_file_disable(
                         return r;
         }
 
-        r = install_context_mark_for_removal(scope, &c, &paths, &remove_symlinks_to, config_path, changes, n_changes);
-        if (r < 0)
-                return r;
+        FOREACH_STRING(config_path, paths.runtime_config, paths.persistent_config) {
```

Comment:
hmm, so i think this change is OK to make, but it should be done comprehensively, i.e. the unmask stuff should also then unmask at both places. And maybe revert and preset need updating for this too?